client_test.go

```diff
@@ -37,6 +37,51 @@ func TestClient(t *testing.T) {
 	assert.Equal(t, []byte("Got message: client testing"), response.Body, "correct body in response")
 }
 
+func TestClientStopWhenCannotStart(t *testing.T) {
+	client := NewClient("amqp://guest:guest@example:1234/")
+
+	assert.NotNil(t, client, "client exist")
+
+	request := NewRequest().
+		WithTimeout(10 * time.Millisecond).
+		WithResponse(false)
+
+	_, err := client.Send(request)
+	assert.Error(t, err)
+
+	var stopped bool
+
+	go func() {
+		client.Stop()
+		stopped = true
+	}()
+
+	assert.Eventually(t, func() bool {
+		return stopped
+	},
+		1*time.Second,
+		10*time.Millisecond,
+	)
+}
+
+func TestClientStopWhenNeverStarted(t *testing.T) {
```

Comment:
This was more of a general test to verify expected behavior when closing a client before initiating the connection to rabbit (on send). Since I will move the toggling of `wantStop` to always be 1 on `Stop()` I could verify that here as well.